### PR TITLE
Multiscreen

### PR DESCRIPTION
**Pull request by Zinglish**
Implemented multiscreen support without breaking the original functionality of the program. Also; I unintentionally modified the XCode stuff too so I'm not sure if that will break compilation or not, hopefully not.

PS: Thanks for the awesome software!

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # osx-wallpaper
 
-> Get or set the desktop wallpaper on OS X
+> Get or set the desktop wallpaper on OS X. Version 1.1.0 supports multiscreen now.
```
Comment (sindresorhus) [inline readme.md]: Should be in the release notes. Not here.

File: readme.md [hunk 2]
```diff
@@ -18,15 +18,27 @@ $ npm install --global osx-wallpaper
 [Download the binary](https://github.com/sindresorhus/osx-wallpaper/releases/latest) and put it in `/usr/local/bin`.
 
 
-## Usage
+## Binary usage
```
Comment (sindresorhus) [inline readme.md]: `Usage` was fine

File: readme.md [hunk 3]
```diff
@@ -18,15 +18,27 @@ $ npm install --global osx-wallpaper
...
 wallpaper unicorn.jpg
 
-# get
+# List connected screens (screen ID:screen name)
+wallpaper --screeninfo
+> 69732482:Color LCD
+> 188845747:SM2333T
+
+# Set wallpaper on specific screen (replace screenid with your screen ID)
+wallpaper screenid unicorn2.jpg
```
Comment (sindresorhus) [inline readme.md]: I'm not sold on using screenid. I was hoping we could do something simple, like just referring to them as screen numbers. Eg, main screen is `1`, the next is `2`, etc. That would also make `--screeninfo` moot. Thoughts?

// @gpapin @dorian-marchal @kevva @arthurvr

File: wallpaper.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -213,7 +220,7 @@
 			isa = XCBuildConfiguration;
 			buildSettings = {
 				CLANG_ENABLE_OBJC_ARC = YES;
-				CODE_SIGN_IDENTITY = "3rd Party Mac Developer Application";
+				CODE_SIGN_IDENTITY = "";
 				PRODUCT_NAME = "$(TARGET_NAME)";
 			};
 			name = Release;
```
Comment (sindresorhus) [inline wallpaper.xcodeproj/project.pbxproj]: Can you drop the changes to unrelated meta files like this and the other ones.

File: wallpaper/ScreenInfo.m [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+	{
+		NSNumber* screenID = [ScreenInfo getScreenDeviceID:screen];
+		if(screenID == deviceID)
+			return screen;
+	}
+	
+	return nil;
+}
+
+@end
```
Comment (sindresorhus) [inline wallpaper/ScreenInfo.m]: If we decide to keep this. Maybe put it up on CocoaPods as a standalone module? Would be much nicer.

File: wallpaper/main.m [hunk 6]
```diff
@@ -5,33 +5,98 @@
...
-		NSArray *args = [NSProcessInfo processInfo].arguments;
+		NSWorkspace *sw  = [NSWorkspace sharedWorkspace];
+		NSArray *args    = [NSProcessInfo processInfo].arguments;
 		NSScreen *screen = [NSScreen mainScreen];
 
 		if (args.count > 1) {
+			// List the version name
 			if ([args[1] isEqualToString: @"--version"]) {
-				puts("1.0.0");
+				puts("1.1.0");
```
Comment (sindresorhus) [inline wallpaper/main.m]: Don't change. I'll do it when releasing.

File: wallpaper/main.m [hunk 7]
```diff
@@ -5,33 +5,98 @@
...
 @import AppKit;
 
+#import "ScreenInfo.h"
+
 int main() {
 	@autoreleasepool {
-		NSWorkspace *sw = [NSWorkspace sharedWorkspace];
-		NSArray *args = [NSProcessInfo processInfo].arguments;
+		NSWorkspace *sw  = [NSWorkspace sharedWorkspace];
+		NSArray *args    = [NSProcessInfo processInfo].arguments;
```
Comment (sindresorhus) [inline wallpaper/main.m]: Don't change code style.

File: wallpaper/main.m [hunk 8]
```diff
@@ -5,33 +5,98 @@
...
+					 "wallpaper on the main screen.\n\n"
+					 
+					 "Usage: wallpaper [options] | [[screenid] [file]]\n"
+					 "   file               Absolute path to image file.\n"
+					 "   screenid           ID of the screen to get or set the wallpaper on\n"
+					 "                      if none specified then the wallpaper is returned or\n"
+					 "                      set on all screens. Use --screeninfo to list screen IDs.\n\n"
+					 
+					 "   --version          Displays version number.\n"
+					 "   --help             Shows this help prompt.\n"
```
Comment (sindresorhus) [inline wallpaper/main.m]: `--version` and `--help` is moot to document. They're always there and implied.

File: wallpaper/main.m [hunk 9]
```diff
@@ -5,33 +5,98 @@
...
 
+			// Print help prompt
 			if ([args[1] isEqualToString: @"--help"]) {
-				puts("\n  Get or set the desktop wallpaper\n\n  Usage: wallpaper [file]\n\n  Created by Sindre Sorhus");
+				puts("Get or set the desktop wallpaper of a specific screen or all screens. If no\n"
+					 "options are specified then OSX Wallpaper will return the path of the current\n"
+					 "wallpaper on the main screen.\n\n"
+					 
+					 "Usage: wallpaper [options] | [[screenid] [file]]\n"
+					 "   file               Absolute path to image file.\n"
```
Comment (sindresorhus) [inline wallpaper/main.m]: Relative paths works too.

**Comment by sindresorhus** [comment]
Nice!

File: readme.md (hunk 3 above)
Comment (kevva) [inline readme.md]: +1, much easier to use too.

Comment (arthurvr) [inline readme.md]: Yep that sounds better :+1:

Comment (Zinglish) [inline readme.md]: Would it make sense to order them according to left to right / top to bottom or just arbitrarily order them?

Comment (sindresorhus) [inline readme.md]: Left to right.

Maybe we should consider only setting the main screen by default (when no argument), to be consistent with the getter. And add a `--all` flag for setting to all screens?

**Comment by sindresorhus** [comment]
Closing for lack of response.
